extras/exporters/gltf-exporter.js

```diff
@@ -101,11 +108,11 @@ class GltfExporter {
                 resources.cameras.push(entity.camera);
             }
 
-            if (entity.render) {
+            if (entity.render && entity.render.enabled) {
                 collectMeshInstances(entity.render.meshInstances);
             }
 
-            if (entity.model) {
+            if (entity.model && entity.model.enabled) {
```

Comment:
```suggestion
            if (entity.model && entity.model.enabled && entity.model.meshInstances) {
```

`ModelComponent#meshInstances` is of type `MeshInstance[]|null`, so this code is crashing currently:

```js
const e = new pc.Entity();
e.addComponent("model");
app.root.addChild(e);
```

Reply:
This is an error which developers trip over all the time, maybe the type `Array|null` should be prevented in the `ModelComponent` itself? I don't really like checking everything to be an array or `null` and everything crashes when its forgotten to check.

Last time I run into this: https://github.com/playcanvas/editor/issues/735

Reply:
thanks, good point.